modules/apps/27-interchain-accounts/depinject.go

```diff
@@ -0,0 +1,98 @@
+package ica
+
+import (
+	"cosmossdk.io/core/appmodule"
+	"cosmossdk.io/depinject"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	modulev1 "github.com/cosmos/ibc-go/api/ibc/applications/interchain_accounts/module/v1"
+	capabilitykeeper "github.com/cosmos/ibc-go/modules/capability/keeper"
+	controllerKeeper "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/controller/keeper"
+	hostKeeper "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/host/keeper"
+	"github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/types"
+	porttypes "github.com/cosmos/ibc-go/v8/modules/core/05-port/types"
+)
+
+var _ depinject.OnePerModuleType = AppModule{}
+
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (AppModule) IsOnePerModuleType() {}
+
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(ProvideModule),
+	)
+}
+
+// ModuleInputs defines the 27-interchain-accounts module inputs for depinject.
+type ModuleInputs struct {
+	depinject.In
+
+	Config *modulev1.Module
+	Cdc    codec.Codec
+	Key    *storetypes.KVStoreKey
+
+	Ics4Wrapper   porttypes.ICS4Wrapper
```

Comment:
this is just a nit, dismiss if you want, hopefully we kill ics4wrapper at some point

```suggestion
	ICS4Wrapper   porttypes.ICS4Wrapper
```